### PR TITLE
Update dependencies to latest versions.

### PR DESCRIPTION
**Pull request by rgrunber**
- Update JDT-LS to 1.14.0 Release
- Update LSP4MP to 0.6.0 SNAPSHOT

Signed-off-by: Roland Grunberg <rgrunber@redhat.com>

### CODE REVIEW

File: quarkus.jdt.ext/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
     <repository>
       <id>jdt.ls.p2</id>
       <layout>p2</layout>
-      <url>https://download.eclipse.org/jdtls/milestones/1.13.0/repository/</url>
+      <url>https://download.eclipse.org/jdtls/milestones/1.14.0/repository/</url>
```
Comment (datho7561) [inline quarkus.jdt.ext/pom.xml:36]: If you are going to bump the referenced JDT-LS version in this PR, I think you should also update it on the qute-ls side (`qute.jdt/pom.xml`)

**Review by datho7561 — CHANGES_REQUESTED** [review]
Other than the comment I left, it looks good and seems to be working.

File: quarkus.jdt.ext/pom.xml (hunk 1 above)
Comment (rgrunber) [reply quarkus.jdt.ext/pom.xml:36]: Thanks. I missed that.

**Comment by rgrunber** [comment]
Copy-paste error :neutral_face:

**Review by datho7561 — APPROVED** [review]

**Pull request merged**